substrate/client/network/src/transport.rs

```diff
@@ -95,12 +130,50 @@ pub fn build_transport(
 		yamux_config
 	};
 
-	let transport = transport
+	transport
 		.upgrade(upgrade::Version::V1Lazy)
 		.authenticate(authentication_config)
 		.multiplex(multiplexing_config)
 		.timeout(Duration::from_secs(20))
-		.boxed();
+}
 
-	transport.with_bandwidth_logging()
+/// Builds the transport that serves as a common ground for all connections.
+///
+/// If `memory_only` is true, then only communication within the same process are allowed. Only
+/// addresses with the format `/memory/...` are allowed.
+///
+/// `yamux_window_size` is the maximum size of the Yamux receive windows. `None` to leave the
+/// default (256kiB).
+///
+/// `yamux_maximum_buffer_size` is the maximum allowed size of the Yamux buffer. This should be
+/// set either to the maximum of all the maximum allowed sizes of messages frames of all
+/// high-level protocols combined, or to some generously high value if you are sure that a maximum
+/// size is enforced on all high-level protocols.
+///
+/// Returns a multiplexed and authenticated implementation of [`libp2p::Transport``].
+pub fn build_transport(
```

Comment:
When we'll be adding rate limiting, we will use `NetworkWorker::new_with_transport` instead of  `NetworkWorker::new`, right? Maybe we can remove `pub` there then?

Reply:
I would prefer to leave it as is - its was a public function of this lib (now it has different type) and it also shows how one can use other methods in this module.

Reply:
changed